zerver/lib/actions.py

```diff
@@ -3252,16 +3252,22 @@ def do_change_tos_version(user_profile: UserProfile, tos_version: str) -> None:
                                  event_type=RealmAuditLog.USER_TOS_VERSION_CHANGED,
                                  event_time=event_time)
 
-def do_regenerate_api_key(user_profile: UserProfile, acting_user: UserProfile) -> None:
-    old_api_key = user_profile.api_key
-    user_profile.api_key = generate_api_key()
-    user_profile.save(update_fields=["api_key"])
-
-    # We need to explicitly delete the old API key from our caches,
-    # because the on-save handler for flushing the UserProfile object
-    # in zerver/lib/cache.py only has access to the new API key.
-    cache_delete(user_profile_by_api_key_cache_key(old_api_key))
-
+def do_regenerate_api_key(user_profile: UserProfile, acting_user: UserProfile) -> str:
```

Comment:
As per conversation with @YagoGG, we might need to make a new method that just regenerate a single API key and replace this method with new one. Also we will need a UI for that first, i am interested in working on that. @timabbott